containerd-shim-spin/src/engine.rs

```diff
@@ -533,10 +552,60 @@ fn trigger_command_for_resolved_app_source(resolved: &ResolvedAppSource) -> Resu
     Ok(trigger_types.iter().map(|x| x.to_string()).collect())
 }
 
+// For each Spin app variable, checks if a container environment variable with
+// the same name exists and duplicates it in the environment with the
+// application variable prefix
+fn configure_application_variables_from_environment_variables(
+    resolved: &ResolvedAppSource,
+) -> Result<()> {
+    let variables: Vec<_> = resolved
+        .variables()
+        .into_iter()
+        .map(|v| v.to_ascii_uppercase())
+        .collect();
+    for variable in variables {
+        let prefixed = format!("{}_{}", SPIN_APPLICATION_VARIABLE_PREFIX, variable);
+        if let Ok(val) = env::var(variable) {
+            env::set_var(prefixed, val);
+        }
+    }
+    Ok(())
+}
```

Comment:
**suggestion**: could use `for_each` to replace the `for` loop for better readability.

```suggestion
fn configure_application_variables_from_environment_variables(
    resolved: &ResolvedAppSource,
) -> Result<()> {
    resolved
        .variables()
        .iter()
        .map(str::to_ascii_uppercase)
        .for_each(|variable| {
            env::var(&variable)
                .ok()
                .map(|val| {
                    let prefixed = format!("{}_{}", SPIN_APPLICATION_VARIABLE_PREFIX, variable);
                    env::set_var(prefixed, val);
                });
        });
    Ok(())
}
```